rspace/src/main/scala/coop/rchain/rspace/history/syntax/HistoryRepositorySyntax.scala

```diff
@@ -0,0 +1,196 @@
+package coop.rchain.rspace.history.syntax
+
+import cats.effect.Sync
+import cats.syntax.all._
+import coop.rchain.rspace.channelStore.{ContinuationHash, DataJoinHash}
+import coop.rchain.rspace.history.HistoryRepository
+import coop.rchain.rspace.syntax.syntaxHistoryRepository
+import coop.rchain.rspace.trace.Event
+import coop.rchain.rspace.trace.Event.{
+  containConflictingEvents,
+  extractJoinedChannels,
+  extractRSpaceEventGroup,
+  produceChannels,
+  Conflict,
+  IsConflict,
+  NonConflict
+}
+import coop.rchain.rspace.{internal, Blake2b256Hash, StableHashProvider}
+import coop.rchain.shared.Serialize
+
+import scala.language.{higherKinds, implicitConversions}
+
+trait HistoryRepositorySyntax {
+  implicit final def syntaxHistoryRepository[F[_]: Sync, C, P, K, A](
+      historyRepo: HistoryRepository[F, C, P, K, A]
+  ): HistoryRepositoryOps[F, C, P, K, A] =
+    new HistoryRepositoryOps[F, C, P, K, A](historyRepo)
+}
+final class HistoryRepositoryOps[F[_]: Sync, C, P, K, A](
+    private val historyRepo: HistoryRepository[F, C, P, K, A]
+) {
+  def getData(state: Blake2b256Hash, channel: C): F[Seq[internal.Datum[K]]] =
+    historyRepo.reset(state) >>= { h =>
+      h.getData(channel)
+    }
+
+  def getJoins(state: Blake2b256Hash, channel: C): F[Seq[Seq[C]]] =
+    historyRepo.reset(state) >>= { h =>
+      h.getJoins(channel)
+    }
+
+  def getContinuations(
+      state: Blake2b256Hash,
+      channels: Seq[C]
+  ): F[Seq[internal.WaitingContinuation[P, A]]] =
+    historyRepo.reset(state) >>= { h =>
+      h.getContinuations(channels)
+    }
+
+  def getData(
+      state: Blake2b256Hash,
+      dataHash: Blake2b256Hash
+  ): F[Seq[internal.Datum[K]]] =
+    historyRepo.reset(state) >>= { h =>
+      h.getData(dataHash)
+    }
+
+  def getJoins(state: Blake2b256Hash, joinHash: Blake2b256Hash): F[Seq[Seq[C]]] =
+    historyRepo.reset(state) >>= { h =>
+      h.getJoins(joinHash)
+    }
+
+  def getContinuations(
+      state: Blake2b256Hash,
+      continuationHash: Blake2b256Hash
+  ): F[Seq[internal.WaitingContinuation[P, A]]] =
+    historyRepo.reset(state) >>= { h =>
+      h.getContinuations(continuationHash)
+    }
+
+  def getDataFromChannelHash(
+      channelHash: Blake2b256Hash
+  ): F[Seq[internal.Datum[K]]] =
+    for {
+      maybeDataHash <- historyRepo.getChannelHash(channelHash)
+      dataHash <- maybeDataHash match {
+                   case Some(DataJoinHash(dataHash, _)) => dataHash.pure[F]
+                   case _ =>
+                     Sync[F].raiseError[Blake2b256Hash](
+                       new Exception(s"not found data hash for $channelHash in channel store")
+                     )
+                 }
```

Comment:
With Option you can just _lift_ it to F if none case is handled as argument of `liftTo`.
```scala
for {
  maybeDataHash <- historyRepo.getChannelHash(channelHash)
  dataHash <- maybeDataHash.liftTo[F](new Exception(s"not found data hash for $channelHash in channel store"))
  ...
}
```

Reply:
I still have to pattern match after `liftTo`. In that case, I think a direct pattern match would be better.

Reply:
The point is to remove pattern matching, `liftTo` is "matching" on None case with an F error as a result. So `dataHash` is the value extracted from Option.
`liftTo` is like a _break_ where code will end if option has no value.

Reply:
`dataHash` can be https://github.com/rchain/rchain/pull/3204/commits/6083e2fefa869fe0a057e849086c879d01bd75ff#diff-d48de69398507c3b8a43f3b7254847b4159df93be5131225e3696ea0a9d79103R16 or https://github.com/rchain/rchain/pull/3204/commits/6083e2fefa869fe0a057e849086c879d01bd75ff#diff-d48de69398507c3b8a43f3b7254847b4159df93be5131225e3696ea0a9d79103R18.
Pattern match is not only for `Option` here.

Reply:
It's just one more _map_ which can select the value and at the end we can decide what to do with _option_. E.g. a vlue inside `DataJoinHash` could be also inside _option_ and in this case we would not need multiple pattern matching.
```scala
maybeDataHash <- historyRepo
  .getChannelHash(channelHash)
  .map(_.map { case DataJoinHash(dataHash, _) => dataHash })
```